pkg/system/path_windows.go

```diff
@@ -2,6 +2,36 @@
 
 package system
 
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
 // DefaultPathEnv is deliberately empty on Windows as the default path will be set by
 // the container. Docker has no context of what the default path should be.
 const DefaultPathEnv = ""
+
+// CheckSystemDriveAndRemoveDriveLetter verifies and manipulates a Windows path.
+// This is used, for example, when validating a user provided path in docker cp.
+// If a drive letter is supplied, it must be the system drive. The drive letter
```

Comment:
Almost, but mixing a couple of things. 

The thing that is being mounted through a -v into a container can be from anywhere on the local system, including outside of the system drive. But correct, we don't support anything but local. The target for the mount inside the container can also be to a drive letter though eg -v c:\foo:d: will give a D drive inside the container. 

What you can't do though is (where this PR comes in) is docker cp where the container path is not the system (c) drive.  Previously, if you used a container path `/foo`, it would have defaulted to the system drive. This PR simply allows users to use a Windows-style path instead on docker cp such as `c:\foo`